src/CachePlugin.php

```diff
@@ -67,12 +68,16 @@ public function handleRequest(RequestInterface $request, callable $next, callabl
 
         if ($cacheItem->isHit()) {
             // return cached response
-            return new FulfilledPromise($cacheItem->get());
+            $data = $cacheItem->get();
+            $response = $data['response'];
+            $response = $response->withBody($this->streamFactory->createStream($data['body']));
+
+            return new FulfilledPromise($response);
         }
 
         return $next($request)->then(function (ResponseInterface $response) use ($cacheItem) {
             if ($this->isCacheable($response)) {
-                $cacheItem->set($response)
+                $cacheItem->set(['response' => $response, 'body' => $response->getBody()->__toString()])
```

Comment:
Thank you for the comments. 

>Concretly we must add something like, if stream is seekable just rewind this, if not buffer it into a new stream.

Hm, okey. I think I understand what you mean.. Give me a minute and I'll update

>> We may face memory problem if big file are downloaded, but don't know if we can avoid that

> sounds like a warning for the cache plugin doc: if you handle large
downloads, make sure the implementation works as expected. or contribute
to make it better. 

Agreed.